spec/log_search_spec.rb

```diff
@@ -36,44 +36,65 @@
+        described_class.new('ReadRegistry'))).length).to eq(7)
     end
 
     it 'does search for either term on the same line' do
       expect(subject.search('TestLogs/test.txt', Or.new(
-                                                            Keyword.new('SmartQueue'),
-                                                            Keyword.new('QueueOfMessages'))).length).to eq(3)
+        described_class.new('SmartQueue'),
+        described_class.new('QueueOfMessages'))).length).to eq(3)
     end
 
     it 'does search for any term from list on the line' do
       expect(subject.search('TestLogs/test.txt',Any.new(
-                                                            [Keyword.new('SmartQueue'),
-                                                             Keyword.new('QueueOfMessages')])).length).to eq(3)
+        [described_class.new('SmartQueue'),
+         described_class.new('QueueOfMessages')])).length).to eq(3)
     end
 
     it 'does search for all terms in a list on the line' do
       expect(subject.search('TestLogs/test.txt',All.new(
-                                                   [Keyword.new('Start'),
-                                                    Keyword.new('New Call'),
-                                                    Keyword.new('Interaction Created')])).length).to eq(5)
+        [described_class.new('Start'),
+         described_class.new('New Call'),
+         described_class.new('Interaction Created')])).length).to eq(5)
     end
 
     it 'does a complicated search' do
       expect(subject.search('TestLogs/test.txt',
-                                     Or.new(
-                                         And.new(
-                                             Keyword.new('INFO'),
-                                             Keyword.new('Utilities')),
-                                         And.new(
-                                             Keyword.new('INFO'),
-                                             Keyword.new('CVHTCall')))).length).to eq(14)
+                            Or.new(
+                              And.new(
+                                described_class.new('INFO'),
+                                described_class.new('Utilities')),
+                              And.new(
+                                described_class.new('INFO'),
+                                described_class.new('CVHTCall')))).length).to eq(14)
+    end
+
+    it 'should return an array containing the correct number of lines' do
+      expect(subject.search('TestLogs/test.txt', described_class.new('uTILITIES',:insensitive)).length).to eq(11)
+    end
+
+    it 'should return an array containing the correct number of lines' do
+      expect(subject.search('TestLogs/test.txt', described_class.new('New Call|Interaction Created',:insensitive)).length).to eq(2)
     end
   end
+end
+
+describe Keyword do
+
+  subject { LogSearch.new }
+
+  it_behaves_like :search_engine
```

Comment:
This is where we verify that the `Keyword` class implements all the require behavior.